### DISCUSSION TITLE
Using Go (WASM) for plugins?

### DISCUSSION THREAD

**Discussion by nikolaydubina** [grafana/grafana#64185] [Plugins]
I have developed fairly useful treemap in Go. http://github.com/nikolaydubina/treemap

Used now in
* http://github.com/nikolaydubina/go-cover-treemap
* http://github.com/nikolaydubina/go-binsize-treemap

I think there are couple useful and interesting applications for Grafana using it.

I would like to implement couple Plugins using this library.

Given that Grafana is developed in backend in Go,

Is there interest in Grafana maintainers to support non-React based Plugins? (eg Go WASM, or some wrappers around compiled JS? or backend driven rendered Plugins (SVG output)?)

Thank you,

**Comment by DemetriusIverslie87L** [comment]
While I cannot speak for the Grafana maintainers, I can say that Grafana does have a plugin development framework that allows for the development of custom plugins using various languages and technologies. However, Grafana plugins are typically developed using React and Typescript, and I am not aware of any official support for non-React based plugins such as those developed with Go WASM or SVG output.

That being said, Grafana is an open-source project and there may be a community of developers interested in non-React based plugins. You may want to explore the Grafana community forums or reach out to the maintainers directly to inquire about the possibility of supporting non-React based plugins.

**Reply by nikolaydubina** [reply]
ok! I thought maintainers are checking this Discussion board

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82847. Feel free to continue the discussion around this there. Thank you!
